src/lib.rs

```diff
@@ -21,149 +75,262 @@
 //! if rng.gen() { // random bool
 //!     let x: f64 = rng.gen(); // random number in range (0, 1)
 //!     println!("x is: {}", x);
+//!     let char = rng.gen::<char>(); // Sometimes you need type annotation
+//!     println!("char is: {}", char);
 //!     println!("Number from 0 to 9: {}", rng.gen_range(0, 10));
 //! }
 //! ```
 //!
-//! The key function is [`Rng::gen()`]. It is polymorphic and so can be used to
-//! generate many types; the [`Standard`] distribution carries the
-//! implementations. In some cases type annotation is required, e.g.
-//! `rng.gen::<f64>()`.
 //!
-//! # Getting random values
+//! ## Random number generators (RNGs)
+//!
+//! Pulling random numbers out of 'thin air' is very hard for a computer. Not
+//! impossible, but hard and slow. Generally the operating systems collects some
+//! of them, by measuring different kinds of noice/jitter in the hardware.
+//!
+//! What is always used instead is a psuedo-random number generator. This is a
+//! clever algorithm that can produce an infinite stream of random numbers from
+//! a small seed. There exist algorithms that produce random numbers that are in
+//! no measurable way any less random than pulling values 'out of thin air'.
+//!
+//! There are two very different groups interested in developing PRNGs: one
+//! interested in simulations and statistics, another in cryptography. From both
+//! groups Rand picks one good PRNG algorithm. They are exposed as:
+//!
+//! - [`SmallRng`]. Uses little memory, and is very fast.
+//!   Note: The currently picked algorithm is not ideal, but in a future version
+//!   the values will for all practical purposes be statistically indiscernible
+//!   from true randomness.
```

Comment:
I don't think this will ever be true for `SmallRng`. Either it is unpredictable and thus cryptographically secure, or we should not claim so. I suggest to not talk about predictability or to say that it is **not** guaranteed to be unpredictable.

Reply:
A "random" source is discernible from true randomness if it is predictable. There's scope for confusion here.